frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationStep/verticalSpatialBoundariesHandler/verticalSpatialBoundariesHandler.component.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useContext, useEffect } from 'react';
+import { Viewer } from '@/v4/services/viewer/viewer';
+import { VIEWER_EVENTS } from '@/v4/constants/viewer';
+import { getDrawingImageSrc } from '@/v5/store/drawings/drawings.helpers';
+import { CalibrationContext } from '../../calibrationContext';
+import { PlaneType, Vector1D } from '../../calibration.types';
+import { TreeActionsDispatchers } from '@/v5/services/actionsDispatchers';
+import { isNull, some } from 'lodash';
+import { ModelHooksSelectors } from '@/v5/services/selectorsHooks';
+import { UNITS_CONVERSION_FACTORS_TO_METRES, addVectors, getTransformationMatrix, getXYPlane, subtractVectors, transformAndTranslate } from '../../calibration.helpers';
+
+export const VerticalSpatialBoundariesHandler = () => {
+	const { verticalPlanes, setVerticalPlanes, vector3D, vector2D, isCalibratingPlanes, setIsCalibratingPlanes, drawingId,
+		setSelectedPlane, selectedPlane, isAlignPlaneActive, setIsAlignPlaneActive } = useContext(CalibrationContext);
+
+	// create element of the drawing to be passed to unity
+	const i = new Image();
+	i.crossOrigin = 'anonymous';
+	i.src = getDrawingImageSrc(drawingId);
+
+	const imageHeight = i.naturalHeight;
+	const imageWidth = i.naturalWidth;
+	
+	const vector3DPlane = getXYPlane(vector3D);
+	const tMatrix = getTransformationMatrix(vector2D, vector3DPlane);
+	const modelUnit = ModelHooksSelectors.selectUnit();
+	
+	useEffect(() => {
+		if (isCalibratingPlanes && !some(verticalPlanes, isNull)) {
+			Viewer.setCalibrationToolMode(isCalibratingPlanes ? 'Vertical' : 'None');
+			Viewer.on(VIEWER_EVENTS.UPDATE_CALIBRATION_PLANES, setVerticalPlanes);
+			return () => {
+				Viewer.setCalibrationToolMode('None');
+				Viewer.off(VIEWER_EVENTS.UPDATE_CALIBRATION_PLANES, setVerticalPlanes);
+			};
+		}
+	}, [isCalibratingPlanes, verticalPlanes]);
+
+	useEffect(() => {
+		Viewer.setCalibrationToolVerticalPlanes(verticalPlanes[0], verticalPlanes[1]);
+	}, [verticalPlanes]);
+
+	useEffect(() => {
+		if (isAlignPlaneActive) {
+			const onPickPoint = ({ position }) => {
+				const initialRange = UNITS_CONVERSION_FACTORS_TO_METRES[modelUnit] * 2.5;
```

Comment:
- I think here you are setting a predefined value for height.
But you should let unity decide that by clicking on a object (@carmenfan am I right?)

take a look at 

`UnityUtil.setCalibrationToolFloorToObject` in here

https://github.com/3drepo/3drepo.io/pull/4858

- I think when entering the plane calibration snapping shouldnt be enabled because the idea is to let the user click on an object first